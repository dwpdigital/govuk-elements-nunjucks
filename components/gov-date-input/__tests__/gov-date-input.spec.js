'use strict';

const nunjucks = require('nunjucks');
const cheerio = require('cheerio');

describe('gov-date-input', () => {
  it('should use ID attribute value over generated IDs', () => {
    const data = {
      legend: 'Date of birth',
      hint: 'For example, 31 3 1980',
      name: 'birth',
      id: 'egg'
    };

    const output = nunjucks.render('./components/gov-date-input/template.njk', data);

    const $ = cheerio.load(output);
    const dayInputId = $('input#egg-day');
    const monthInputId = $('input#egg-month');
    const yearInputId = $('input#egg-year');
    const dayLabelFor = $('label[for="egg-day"]');
    const monthLabelFor = $('label[for="egg-month"]');
    const yearLabelFor = $('label[for="egg-year"]');
    expect(dayInputId, 'Day ID not found');
    expect(monthInputId, 'Month ID not found');
    expect(yearInputId, 'Year ID not found');
    expect(dayLabelFor, 'Matching label for Day not found');
    expect(monthLabelFor, 'Matching label for Month not found');
    expect(yearLabelFor, 'Matching label for Year not found');
    expect(output).toMatchSnapshot();
  });
  it('should use the maxYear attr to set the max attr on year input', () => {
    const data = {
      legend: 'Date of birth',
      hint: 'For example, 31 3 1980',
      name: 'birth',
      maxYear: '2020'
    };

    const output = nunjucks.render('./components/gov-date-input/template.njk', data);
    const $ = cheerio.load(output);
    const max = $('#input-birth-year').attr('max');

    expect(max).toBe(data.maxYear);
    expect(output).toMatchSnapshot();
  });

  it('should set the values of the inputs using the value attribute', () => {
    const data = {
      name: 'birth',
      legend: 'Date of birth',
      hint: 'For example, 31 3 1980',
      value: {day: '12', month: '12', year: '2015'}
    };

    const output = nunjucks.render('./components/gov-date-input/template.njk', data);
    const $ = cheerio.load(output);
    const dayValue = $('#input-birth-day').attr('value');
    const monthValue = $('#input-birth-month').attr('value');
    const yearValue = $('#input-birth-year').attr('value');

    expect(dayValue).toBe(data.value.day);
    expect(monthValue).toBe(data.value.month);
    expect(yearValue).toBe(data.value.year);
    expect(output).toMatchSnapshot();
  });

  it('should add a form hint using the hint attribute', () => {
    const data = {
      legend: 'Date of birth',
      hint: 'For example, 31 3 1980',
      name: 'birth'
    };

    const output = nunjucks.render('./components/gov-date-input/template.njk', data);
    const $ = cheerio.load(output);

    const formHintText = $('legend span.form-hint').html();

    expect(formHintText).toBe(data.hint);
    expect(output).toMatchSnapshot();
  });

  it('should use the maxYear attr to set the max attr on year input', () => {
    const data = {
      legend: 'Date of birth',
      hint: 'For example, 31 3 1980',
      name: 'birth',
      maxYear: '2020'
    };

    const output = nunjucks.render('./components/gov-date-input/template.njk', data);
    const $ = cheerio.load(output);
    const max = $('#input-birth-year').attr('max');

    expect(max).toBe(data.maxYear);
    expect(output).toMatchSnapshot();
  });
});
