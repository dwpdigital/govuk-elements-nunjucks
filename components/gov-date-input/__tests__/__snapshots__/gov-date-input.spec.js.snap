// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gov-date-input should add a form hint using the hint attribute 1`] = `
<div id="input-birth-form" class="form-group">
  <fieldset>
    <div class="form-date">
      <legend>
        <span class="form-label-bold">Date of birth</span>
        <span class="form-hint">For example, 31 3 1980</span>
      </legend>
      <div class="form-group form-group-day">
        <label for="input-birth-day">Day</label>
        <input class="form-control" id="input-birth-day" name="birthDay"
        value="" type="number" pattern="[0-9]*" min="0" max="31" maxlength="2"
        />
      </div>
      <div class="form-group form-group-month">
        <label for="input-birth-month">Month</label>
        <input class="form-control" id="input-birth-month" name="birthMonth"
        value="" type="number" pattern="[0-9]*" min="0" max="31" maxlength="2"
        />
      </div>
      <div class="form-group form-group-year">
        <label for="input-birth-year">Year</label>
        <input class="form-control" id="input-birth-year" name="birthYear"
        value="" type="number" pattern="[0-9]*" min="0" maxlength="4" />
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`gov-date-input should set the values of the inputs using the value attribute 1`] = `
<div id="input-birth-form" class="form-group">
  <fieldset>
    <div class="form-date">
      <legend>
        <span class="form-label-bold">Date of birth</span>
        <span class="form-hint">For example, 31 3 1980</span>
      </legend>
      <div class="form-group form-group-day">
        <label for="input-birth-day">Day</label>
        <input class="form-control" id="input-birth-day" name="birthDay"
        value="12" type="number" pattern="[0-9]*" min="0" max="31" maxlength="2"
        />
      </div>
      <div class="form-group form-group-month">
        <label for="input-birth-month">Month</label>
        <input class="form-control" id="input-birth-month" name="birthMonth"
        value="12" type="number" pattern="[0-9]*" min="0" max="31" maxlength="2"
        />
      </div>
      <div class="form-group form-group-year">
        <label for="input-birth-year">Year</label>
        <input class="form-control" id="input-birth-year" name="birthYear"
        value="2015" type="number" pattern="[0-9]*" min="0" maxlength="4" />
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`gov-date-input should use ID attribute value over generated IDs 1`] = `
<div id="egg-form" class="form-group">
  <fieldset>
    <div class="form-date">
      <legend>
        <span class="form-label-bold">Date of birth</span>
        <span class="form-hint">For example, 31 3 1980</span>
      </legend>
      <div class="form-group form-group-day">
        <label for="egg-day">Day</label>
        <input class="form-control" id="egg-day" name="birthDay" value=""
        type="number" pattern="[0-9]*" min="0" max="31" maxlength="2" />
      </div>
      <div class="form-group form-group-month">
        <label for="egg-month">Month</label>
        <input class="form-control" id="egg-month" name="birthMonth"
        value="" type="number" pattern="[0-9]*" min="0" max="31" maxlength="2"
        />
      </div>
      <div class="form-group form-group-year">
        <label for="egg-year">Year</label>
        <input class="form-control" id="egg-year" name="birthYear"
        value="" type="number" pattern="[0-9]*" min="0" maxlength="4" />
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`gov-date-input should use the maxYear attr to set the max attr on year input 1`] = `
<div id="input-birth-form" class="form-group">
  <fieldset>
    <div class="form-date">
      <legend>
        <span class="form-label-bold">Date of birth</span>
        <span class="form-hint">For example, 31 3 1980</span>
      </legend>
      <div class="form-group form-group-day">
        <label for="input-birth-day">Day</label>
        <input class="form-control" id="input-birth-day" name="birthDay"
        value="" type="number" pattern="[0-9]*" min="0" max="31" maxlength="2"
        />
      </div>
      <div class="form-group form-group-month">
        <label for="input-birth-month">Month</label>
        <input class="form-control" id="input-birth-month" name="birthMonth"
        value="" type="number" pattern="[0-9]*" min="0" max="31" maxlength="2"
        />
      </div>
      <div class="form-group form-group-year">
        <label for="input-birth-year">Year</label>
        <input class="form-control" id="input-birth-year" name="birthYear"
        value="" type="number" pattern="[0-9]*" min="0" max="2020" maxlength="4"
        />
      </div>
    </div>
  </fieldset>
</div>
`;
